fontbe/src/glyphs.rs

```diff
@@ -32,6 +32,37 @@ pub fn create_glyf_work(glyph_name: GlyphName) -> Box<BeWork> {
     Box::new(GlyphWork { glyph_name })
 }
 
+/// Can glyph reuse the metrics of other?
+///
+/// To be safe the component should have:
+///
+/// * The same advance width as glyph
+/// * A 2x2 transform that does nothing (basis vectors do not change)
+/// * No x-translation
+///    * y-translation is OK
+///
+/// This forces the composite glyph to use the possibly hinted horizontal
+/// metrics of the sub-glyph, instead of those from the "hmtx" table.
+///
+/// See <https://github.com/googlefonts/ufo2ft/blob/0c0a570b84d1351ab704ba1fa5ae03aeef51179f/Lib/ufo2ft/instructionCompiler.py#L151-L173>
+fn can_reuse_metrics(glyph: &ir::Glyph, component_glyph: &ir::Glyph, transform: &Affine) -> bool {
+    for (loc, inst) in glyph.sources().iter() {
+        // We won't stress whether the interpolated values would match just yet
+        let Some(other_inst) = component_glyph.sources().get(loc) else {
+            return false;
+        };
```

Comment:
ufo2ft outlineCompiler works on single master UFOs when setting that flag on each master glyf table, which are then later on used by fontTools.varLib to generate gvar deltas. Note how in fonttools the component flags (like USE_MY_METRICS) are not even considered when determining whether components from different masters can actually interpolate (`_GlyphControls.flags` are None for composite glyphs):

https://github.com/fonttools/fonttools/blob/1c283756a5e39d69459eea80ed12792adc4922dd/Lib/fontTools/ttLib/tables/_g_l_y_f.py#L400

which means that whatever was in the default instance's glyf table is going to also end up in the final VF, since fontTools.varLib basically starts by cloing the default master TTF and tacking onto it the variation tables.

Reply:
given that glyf component flags cannot "vary" (unlike, say, dx,dy translation), the expectation should be that they are identical across all masters; so if any of the masters are such that the USE_MY_METRICS flag would not be set, the default instance of the VF should not force that flag on the entire variation space. So arguably your code is more correct and I will file an issue in fonttools about this.

Reply:
To close the loop, https://github.com/fonttools/fonttools/issues/3085.

Reply:
(so I will not change the code here)